pkg/operator/controller/status.go

```diff
@@ -40,71 +46,153 @@ func (r *reconciler) syncOperatorStatus() error {
 			return fmt.Errorf("failed to get clusteroperator %s: %v", co.Name, err)
 		}
 	}
+	oldStatus := co.Status.DeepCopy()
 
-	ns, ingresses, err := r.getOperatorState()
+	ingresses, err := r.getOperatorState(ns.Name)
```

Comment:
`computeOperatorFailingCondition` is expecting `ns` to be nil if the namespace does not exist.  With this change, the `if ns ==n nil` in `computeOperatorFailingCondition` becomes dead code.

Reply:
fixed